Forge.TreeWalker/src/ITreeWalkerCallbacksV2.cs

```diff
@@ -0,0 +1,31 @@
+//-----------------------------------------------------------------------
+// <copyright file="ITreeWalkerCallbacksV2.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The ITreeWalkerCallbacksV2 interface. Use TreeNodeContext to wrap all the required input and output info.
+// </summary>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Forge.TreeWalker
+{
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// The ITreeWalkerCallbacks interface defines the callback Tasks that are awaited while walking the tree.
+    /// </summary>
+    public interface ITreeWalkerCallbacksV2
+    {
+        /// <summary>
+        /// The callback Task that is awaited before visiting each node.
+        /// </summary>
+        /// <param name="treeNodeContext">The TreeNode context holding relevant information for this Action.</param>
```

Comment:
>The TreeNode context holding relevant information for this Action. [](start = 42, length = 66)

The tree node context holding relevant information about this tree node and session.

Reply:
updated. applied the same to AfterVisitNode() below.

---
In reply to: [506633634](https://github.com/microsoft/Forge/pull/32#discussion_r506633634) [](ancestors = 506633634)